tests/unit/abilities-api/wpAbilitiesRegistry.php

```diff
@@ -427,4 +427,115 @@ public function test_wp_ability_invalid_properties_throws_exception() {
 			)
 		);
 	}
+
+	/**
+	 * Test register_ability_args filter modifies the args before ability instantiation.
+	 */
+	public function test_register_ability_args_filter_modifies_args() {
+		$was_filter_callback_fired = false;
+
+		// Define the filter.
+		$filter_callback = static function ( $args ) use ( &$was_filter_callback_fired ) {
+			$args['label']             = 'Modified label';
+			$original_execute_callback = $args['execute_callback'];
+			$args['execute_callback']  = static function ( array $input ) use ( &$was_filter_callback_fired, $original_execute_callback ) {
+				$was_filter_callback_fired = true;
+				return $original_execute_callback( $input );
+			};
+
+			return $args;
+		};
+		add_filter( 'register_ability_args', $filter_callback, 10 );
+
+		// Register the ability.
+		$ability = $this->registry->register( self::$test_ability_name, self::$test_ability_args );
+
+		// Check the label was modified by the filter.
+		$this->assertSame( 'Modified label', $ability->get_label() );
+
+		// Call the execute callback.
+		$result = $ability->execute(
+			array(
+				'a' => 1,
+				'b' => 2,
+			)
+		);
+
+		$this->assertTrue( $was_filter_callback_fired, 'The execute callback defined in the filter was not fired.' );
+		$this->assertSame( 3, $result, 'The original execute callback did not return the expected result.' );
+
+		// Clean up by removing the filter.
+		remove_filter( 'register_ability_args', $filter_callback, 10 );
```

Comment:
Nitpick, here in other tests, it'd be safer to follow WP core and move test cleanup before assertion to ensure they always execute even when the assertion fails.

Reply:
https://github.com/WordPress/abilities-api/pull/74/commits/36c0e992a259af669eb8bd717985a552de3cb04f